grpc/binlog/v1alpha/binarylog.proto

```diff
@@ -35,7 +35,31 @@ message GrpcLogEntry {
     RECV_INITIAL_METADATA = 4;
     RECV_TRAILING_METADATA = 5;
     RECV_MESSAGE = 6;
-  };
+  }
+
+  // Enumerates the type of logs
+  enum EventType {
+    UNKNOWN_EVENT_TYPE = 0;
+    // Headers sent from client to server
+    REQUEST_HEADERS = 1;
+    // Headers sent from server to client
+    RESPONSE_HEADERS = 2;
+    // Message sent from client to server
+    REQUEST_MESSAGE = 3;
+    // message sent from server to client
+    RESPONSE_MESSAGE = 4;
+    // Signal that client is done sending
+    HALF_CLOSE = 5;
+    // Trailers sent from server to client.
+    // On client side, this is the final entry for an RPC.
+    // On server side, a CANCEL can still happen after this.
+    STATUS_TRAILERS = 6;
+    // Signal that the RPC should be cancelled
+    CANCEL = 7;
+    // A special event that only applicable to server side.
```

Comment:
Have client-side generate this too. It's trival for the client logger to do and it makes the log completely consistent between client and server.

Reply:
We can be consistent in the sense that the end of RPC is always marked with the end marker. But of course, the two side can still see different things in the event of a failure (client sees: `...> STATUS_TRAILERS, END_MARKER` but server sees: `... STATUS_TRAILERS, CANCEL, END_MARKER`).